examples/nidaqmx/counter-output.py

```diff
@@ -50,45 +50,34 @@
 task = None
 
 
-def raise_if_error(response):
-    """Raise an exception if an error was returned."""
-    if response.status != 0:
-        response = client.GetErrorString(
-            nidaqmx_types.GetErrorStringRequest(error_code=response.status)
-        )
-        raise Exception(f"Error: {response.error_string}")
-
-
 try:
     response = client.CreateTask(nidaqmx_types.CreateTaskRequest(session_name="my task"))
-    raise_if_error(response)
     task = response.task
 
-    raise_if_error(
-        client.CreateCOPulseChanTime(
-            nidaqmx_types.CreateCOPulseChanTimeRequest(
-                task=task,
-                counter=COUNTER_NAME,
-                units=nidaqmx_types.DIGITAL_WIDTH_UNITS3_SECONDS,
-                idle_state=nidaqmx_types.LEVEL1_LOW,
-                initial_delay=1.0,
-                low_time=0.5,
-                high_time=1.0,
-            )
+    client.CreateCOPulseChanTime(
+        nidaqmx_types.CreateCOPulseChanTimeRequest(
+            task=task,
+            counter=COUNTER_NAME,
+            units=nidaqmx_types.DIGITAL_WIDTH_UNITS3_SECONDS,
+            idle_state=nidaqmx_types.LEVEL1_LOW,
+            initial_delay=1.0,
+            low_time=0.5,
+            high_time=1.0,
         )
     )
 
-    raise_if_error(client.StartTask(nidaqmx_types.StartTaskRequest(task=task)))
+    client.StartTask(nidaqmx_types.StartTaskRequest(task=task))
```

Comment:
@bkeryan what DAQ methods ought we check for warnings?

Reply:
I think the most important ones are StartTask and the various Read/Write methods.

Hardware timed AO and DO often uses IsTaskDone or WaitUntilTaskDone to check for errors and warnings, but I don't think these examples demonstrate these use cases.

StopTask can return DAQmxWarningStoppedBeforeDone (200010). It would be reasonable to move StopTask out of the finally block and check for warnings. ClearTask will do all necessary cleanup, including stopping the task, but I don't think it reports errors or warnings that occur during cleanup.